lib/state/app_state.dart

```diff
@@ -11,6 +12,7 @@ class AppState with _$AppState {
     //TODO: this will be removed
     @Default('') String sample,
     @Default(<Pokemon>[]) List<Pokemon> pokemons,
+    @Default(<PokemonDetails>[]) List<PokemonDetails> pokemonDetails,
```

Comment:
why does it need to be a list? I suggest make it a nullable PokemonDetails then provide a null value as its default 🙂